airplane-mode.php

```diff
@@ -529,6 +529,38 @@ public function toggle_css() {
 			wp_enqueue_style( 'airplane-mode', plugins_url( '/lib/css/' . $file, __FILE__ ), array(), $vers, 'all' );
 		}
 
+		public function set_mode( $mode = 'on' ) {
+			if ( ! in_array( $mode, array( 'on', 'off' ) ) ) {
+				$mode = 'on';
+			}
+
+			// Update the setting.
+			$return = update_site_option( 'airplane-mode', $mode );
+
+			// Fire action to allow for functions to run on status change.
+			do_action( 'airplane_mode_status_change', $mode );
+
+			return $return;
```

Comment:
@afragen It will return true if the value was updated, and false if it wasn't (or couldn't be). I generally return this as a best practice. Cases where you might use this:

- Needing to know if the thing changed vs wast just set to the same value as before.
- If you know it should have changed, and it didn't, you could know that something went really wrong, like a database failure, or such.

I captured the return value because I didn't want to move the `do_action()` call relative to the option update, for back compat.

I could see an argument for having it return the boolean state, but it also might be weird if you call `set_mode( 'off' )` and it returns `false` when the command was successful in changing the mode from 'on' to 'off'.